src/main/java/com/atlassian/db/replica/internal/ReplicaStatement.java

```diff
@@ -94,7 +95,8 @@ public void close() throws SQLException {
     @Override
     public int getMaxFieldSize() throws SQLException {
         checkClosed();
-        return getWriteStatement(new RouteDecisionBuilder(RW_API_CALL)).getMaxFieldSize();
+        //todo this looks to be READ operation but run on write statement?
```

Comment:
It's easier to track it with issues than todos (see https://github.com/atlassian-labs/db-replica/issues/37 for example)

Reply:
Thanks for the suggestion.
TBH I wasn't feeling confident about creating issues, and just created `draft` PR with bunch of comments to share my concerns and get feedback whether they're relevant.

Reply:
No worries. The draft was a good call. It engaged multiple people from the team. I guess the final PR will benefit from this early draft. We can create the issues not to lose them or fix them in the PR (I guess it's outside the scope).